### PR TITLE
GH-1842 Generalize WsECL connection

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes GH-1842

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage @mlzummo Please Review.

File: src/other/Comms.js [hunk 1]
```diff
@@ -374,25 +374,14 @@
...
-                    queryParts = pathParts[1].split("/");
-                    if (queryParts.length >= 3) {
-                        this.target(queryParts[1]);
-                        this.query(queryParts[2]);
-                    }
-                }
-            }
-            if (!this._target && !this._query) {
+            if (!this._target || !this._query) {
                 //http://10.241.100.157:8002/WsEcl/forms/default/query/roxie/wecare
```
Comment (jchambers-ln) [inline src/other/Comms.js]: commented out reference to an internal IP I would revome

File: src/other/Comms.js [hunk 2]
```diff
@@ -374,25 +374,14 @@
...
-                if (pathParts.length >= 2) {
-                    queryParts = pathParts[1].split("/");
-                    if (queryParts.length >= 3) {
-                        this.target(queryParts[1]);
-                        this.query(queryParts[2]);
-                    }
-                }
-            }
-            if (!this._target && !this._query) {
+            if (!this._target || !this._query) {
```
Comment (jchambers-ln) [inline src/other/Comms.js]: changing from an && to an !! means this.query will be evaluated even if this._target is not could this introduce a null pointer?

**Comment by jchambers-ln** [comment]
see in line, otherwise good-- not sure if changes are needed but called out the change in Boolean statement.

File: src/other/Comms.js (hunk 2 above)
Comment (GordonSmith) [inline src/other/Comms.js]: Its the other way round - if either are null then I want to try an alternative parse.

**Comment by GordonSmith** [comment]
@jchambers-ln Changes made as suggested.

**Pull request merged**